core/src/com/recursive/corruption/gravoball/Player.java

```diff
@@ -4,17 +4,80 @@
 import com.badlogic.gdx.graphics.Color;
 import com.badlogic.gdx.math.Vector2;
 
+
+
+
 public class Player extends Ball {
+
+    private int health = 100;
+
+    private boolean isDead = false;
+
     Player(float x, float y) {
-        super(x, y, 6.f, 1000.f, new Color(1.0f, .1f, .1f, 1.0f));
+        super(x, y, 6.f, 1.f, new Color(1.0f, .1f, .1f, 1.0f));
+
     }
 
     public void updateBall(GravoBallGame game, float playerDt) {
         pos.x = Gdx.input.getX();
         pos.y = Gdx.graphics.getHeight() - Gdx.input.getY();
+
+        // regen health every frame
+        if (!isDead && health < 100) {
+            health++;
+        }
+
+        //gain mass every frame exponentially - makes a dramatic start
+        if (mass < 250) {
+            mass *= 1.027;
+            mass += 0.25;
+        } else {
+            mass += 0.5;
+        }
+        //System.out.println("Mass: " + mass);
+
+        getLivingStatus();
+
         game.scorePoints(playerDt);
     }
 
+    // TODO death GUI update
+    // Am I alive?
+    private boolean getLivingStatus() {
```

Comment:
Since this updates internal state, we should probably call it `updateLivingStatus` or something like that. And it shouldn't need to return anything since if we wanted a real getter we could just have `boolean isAlive() { return !isDead; }`.